### DISCUSSION TITLE
Welcome to Fooocus-API Discussions!

### DISCUSSION THREAD

**Discussion by mrhan1993** [mrhan1993/Fooocus-API#312] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bobgus39** [comment]
how is the json structure to insert loras?

**Reply by mrhan1993** [reply]
just like this
```python
{
  "prompt": "",
  "negative_prompt": "",
  "style_selections": [
    "Fooocus V2",
    "Fooocus Enhance",
    "Fooocus Sharp"
  ],
  "performance_selection": "Speed",
  "aspect_ratios_selection": "1152*896",
  "image_number": 1,
  "image_seed": -1,
  "sharpness": 2,
  "guidance_scale": 4,
  "base_model_name": "juggernautXL_v8Rundiffusion.safetensors",
  "refiner_model_name": "None",
  "refiner_switch": 0.5,
  "loras": [
    {
      "enabled": true,
      "model_name": "sd_xl_offset_example-lora_1.0.safetensors",
      "weight": 0.1
    },
    {
      "enabled": true,
      "model_name": "None",
      "weight": 1
    },
    {
      "enabled": true,
      "model_name": "None",
      "weight": 1
    },
    {
      "enabled": true,
      "model_name": "None",
      "weight": 1
    },
    {
      "enabled": true,
      "model_name": "None",
      "weight": 1
    }
  ],
  "advanced_params": {
    "adaptive_cfg": 7,
    "adm_scaler_end": 0.3,
    "adm_scaler_negative": 0.8,
    "adm_scaler_positive": 1.5,
    "canny_high_threshold": 128,
    "canny_low_threshold": 64,
    "controlnet_softness": 0.25,
    "debugging_cn_preprocessor": false,
    "debugging_inpaint_preprocessor": false,
    "disable_intermediate_results": false,
    "disable_preview": false,
    "disable_seed_increment": false,
    "freeu_b1": 1.01,
    "freeu_b2": 1.02,
    "freeu_enabled": false,
    "freeu_s1": 0.99,
    "freeu_s2": 0.95,
    "inpaint_disable_initial_latent": false,
    "inpaint_engine": "v2.6",
    "inpaint_erode_or_dilate": 0,
    "inpaint_mask_upload_checkbox": false,
    "inpaint_respective_field": 1,
    "inpaint_strength": 1,
    "invert_mask_checkbox": false,
    "mixing_image_prompt_and_inpaint": false,
    "mixing_image_prompt_and_vary_upscale": false,
    "overwrite_height": -1,
    "overwrite_step": -1,
    "overwrite_switch": -1,
    "overwrite_upscale_strength": -1,
    "overwrite_vary_strength": -1,
    "overwrite_width": -1,
    "refiner_swap_method": "joint",
    "sampler_name": "dpmpp_2m_sde_gpu",
    "scheduler_name": "karras",
    "skipping_cn_preprocessor": false
  },
  "save_meta": true,
  "meta_scheme": "fooocus",
  "save_extension": "png",
  "save_name": "",
  "read_wildcards_in_order": false,
  "require_base64": false,
  "async_process": false,
  "webhook_url": ""
}
```

**Comment by bobgus39** [comment]
hi, 

i'm trying to replicate the inpaint-outpaint and I have that photo with its mask
![Aitana-338-2](https://github.com/mrhan1993/Fooocus-API/assets/126803536/5b86e432-b82f-49f8-93e7-6160c6dc2dbe)
![descarga](https://github.com/mrhan1993/Fooocus-API/assets/126803536/8c087937-be09-4d27-9ee5-c898095fe69a)

and I pass with that parametres:
curl -X 'POST' \
  'http://127.0.0.1:8888/v1/generation/image-inpaint-outpaint' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'sharpness=2' \
  -F 'input_mask=@descarga.png;type=image/png' \
  -F 'outpaint_distance_right=0' \
  -F 'loras=[{"enabled":true,"model_name":"sd_xl_offset_example-lora_1.0.safetensors","weight":0.1},{"enabled":true,"model_name":"None","weight":1.0},{"enabled":true,"model_name":"None","weight":1.0},{"enabled":true,"model_name":"None","weight":1.0},{"enabled":true,"model_name":"None","weight":1.0}]' \
  -F 'outpaint_distance_left=0' \
  -F 'advanced_params=' \
  -F 'read_wildcards_in_order=false' \
  -F 'guidance_scale=4' \
  -F 'prompt=blond hairstyle' \
  -F 'input_image=@Aitana-338-2.jpg;type=image/jpeg' \
  -F 'outpaint_distance_bottom=0' \
  -F 'save_meta=false' \
  -F 'webhook_url=' \
  -F 'require_base64=false' \
  -F 'async_process=false' \
  -F 'save_name=' \
  -F 'image_number=4' \
  -F 'negative_prompt=' \
  -F 'refiner_switch=0.5' \
  -F 'base_model_name=juggernautXL_v8Rundiffusion.safetensors' \
  -F 'image_seed=-1' \
  -F 'style_selections=Fooocus V2,Fooocus Enhance,Fooocus Sharp' \
  -F 'inpaint_additional_prompt=blonde hair' \
  -F 'outpaint_selections=' \
  -F 'outpaint_distance_top=0' \
  -F 'refiner_model_name=None' \
  -F 'aspect_ratios_selection=896*1152' \
  -F 'performance_selection=Speed' \
  -F 'save_extension=png' \
  -F 'meta_scheme=fooocus'

that's the result:
![62a30326-6658-4815-9467-e1af6c4b1ee3-3](https://github.com/mrhan1993/Fooocus-API/assets/126803536/7e06cbf8-6b3a-41f2-bb89-2e18dbc0b246)

Somebody knows why the image with its mask is not working?

**Reply by bobgus39** [reply]
the error is in the contrast, you need a mask like this:
![maskBgBlack](https://github.com/mrhan1993/Fooocus-API/assets/126803536/a355bb24-2727-4ef8-bfb6-6bbcbd82cff2)

and thats the result with prompt blonde hairstyle:
![bcba0c8d-ad71-40d3-a0fe-47bb609aa2ea-3](https://github.com/mrhan1993/Fooocus-API/assets/126803536/08388cd6-b133-4cd2-ab33-0868690b4020)

**Comment by ijash** [comment]
so, the mask basically :

**white** = area that will be generated, 
**black** = area preserved

is that correct?

**Reply by mrhan1993** [reply]
yes
